options/ansi/generic/string-stubs.cpp

```diff
@@ -414,7 +414,144 @@ int strverscmp(const char *l0, const char *r0) {
 	return l[i] - r[i];
 }
 
-void *memmem(const void *, size_t, const void *, size_t) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+// Taken from musl.
+static char *twobyte_memmem(const unsigned char *h, size_t k, const unsigned char *n) {
+	uint16_t nw = n[0]<<8 | n[1], hw = h[0]<<8 | h[1];
```

Comment:
There's a _lot_ of these few character variables, and it makes the code quite hard to read. Please rename them to be more readable.